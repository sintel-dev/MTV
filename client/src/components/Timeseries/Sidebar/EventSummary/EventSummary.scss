@import '../../../../assets/sass/form.scss';
.event-summary{
    text-transform: uppercase;
    width: 100%;
    box-sizing: border-box;
    border: 1px solid #60666C;
    padding: 16px 0 0 0;
    border-radius: 4px;
    font-size: 12px;
    background: #26282E;


    .event-header {
        padding: 0 16px;
    }
    .tooltip-data {
        font-size: 12px;
        background: rgba(103, 106, 130, 0.8);
        padding: 10px;
        border-radius: 4px;
        border-bottom-left-radius: 0;
        position: absolute;
        z-index: 999;
        color: #fff;
        text-transform: capitalize;
        top: -37px;
        left: 0;
        white-space: nowrap;
        display: none;
        transition: all .2s ease-in-out;
    }
    table{
        border-collapse: collapse;
        width: 100%;
        th {
            text-align: left;
            font-weight: normal;
            padding: 0 5px 0 16px;
        }
        td{
            text-align: center;
            height: 38px;
            position: relative;
            &.highlighted{
                background-color: rgba(35, 36, 43, 0.5);
                cursor: pointer;
                .tooltip-data{
                    display: block;
                }
            }

            &:last-child{
                .tooltip-data{
                    left: auto;
                    right: 0;
                    border-bottom-left-radius: 3px;
                    border-bottom-right-radius: 0;
                }
            }
        }
        .row-light{
            background: #2C3034
        }

        tr:last-child{
            th{
                border-bottom-left-radius: 4px;
            }
            td{
                &:last-child{
                    border-bottom-right-radius: 4px;
                }
            }
        }
    }

    .event-header {
        height: 40px;
        line-height: 27px;
        .left-wrapper{
            float: left;
        }
        .right-wrapper{
            float: right
        }
        button{
            background: transparent;
            border: none;
            color: #fff;
            cursor: pointer;
            outline: none;
            span{
                display: inline-block;
                margin-right: 10px;
            }
        }
    }

    .summary-details {
        display: none;
        &.active{
            display: block;
        }
        .indicator{
            width: 7px;
            height: 7px;
            margin: 0 10px;
        }
    }

}