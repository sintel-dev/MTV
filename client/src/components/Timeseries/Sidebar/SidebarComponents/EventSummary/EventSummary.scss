@import '../../../../../assets/sass/form.scss';
.event-summary {
    text-transform: uppercase;
    width: 100%;
    box-sizing: border-box;
    padding: 5px 0 0 0;
    margin: 0 0 15px 0;
    border-radius: 4px;
    font-size: 12px;
    border-bottom: 1px solid #23242b;

    .event-header {
        margin: 0 16px;
    }

    .tooltip-data {
        font-size: 12px;
        background: rgba(103, 106, 130, 0.8);
        padding: 10px;
        border-radius: 4px;
        border-bottom-left-radius: 0;
        position: absolute;
        z-index: 999;
        color: #fff;
        text-transform: capitalize;
        top: -37px;
        left: 0;
        white-space: nowrap;
        display: none;
        transition: all 0.2s ease-in-out;
    }
    table {
        border-collapse: collapse;
        width: 100%;
        th {
            text-align: left;
            font-weight: normal;
            padding: 0 5px 0 16px;
        }
        td {
            text-align: center;
            height: 38px;
            position: relative;
            &.highlighted {
                background-color: rgba(35, 36, 43, 0.5);
                cursor: pointer;
                .tooltip-data {
                    display: block;
                }
            }

            &:last-child {
                .tooltip-data {
                    left: auto;
                    right: 0;
                    border-bottom-left-radius: 3px;
                    border-bottom-right-radius: 0;
                }
            }
        }

        tr:last-child {
            th {
                border-bottom-left-radius: 4px;
            }
            td {
                &:last-child {
                    border-bottom-right-radius: 4px;
                }
            }
        }
    }

    .event-header {
        height: 40px;
        line-height: 27px;
        display: flex;
        align-items: center;
        justify-content: space-between;

        .left-wrapper {
            font-size: 20px;
            font-weight: bold;
        }
        li {
            margin-right: 20px;
            &:last-child {
                margin-right: 0;
            }
        }
    }

    .summary-details {
        border-top: 1px solid #23242b;
        border-width: 90%;

        .indicator {
            width: 7px;
            height: 7px;
            margin: 0 10px;
        }
    }
}
