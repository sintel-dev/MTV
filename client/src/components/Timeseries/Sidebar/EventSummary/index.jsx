import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { tagSeq, fromTagToClassName } from '../../../Landing/utils';
import { fromMonthToIndex } from '../../../../model/utils/Utils';

import './EventSummary.scss';

const renderTagIcon = () =>
  tagSeq.map(currentTag => (
    <td key={currentTag}>
      <span className="tooltip-data">
        <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
        {currentTag}
      </span>
      <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
    </td>
  ));

const countEventsPerTag = (tag, events) => {
  const currentEvents = Object.values(events);
  return currentEvents.filter(currentEvent => currentEvent.tag === tag).length;
};

const renderTagEvents = (isPeriodLevelSelected, grouppedEvents) => {
  if (!isPeriodLevelSelected || grouppedEvents === undefined) {
    return tagSeq.map(currentTag => <td key={currentTag}>-</td>);
  }
  return tagSeq.map(currentTag => <td key={currentTag}>{countEventsPerTag(currentTag, grouppedEvents.events)}</td>);
};

const renderTagEventsPerMonth = (isPeriodLevelSelected, month, monthEvents) => {
  if (!isPeriodLevelSelected || month === '' || monthEvents === undefined) {
    return tagSeq.map(currentTag => <td key={currentTag}>-</td>);
  }
  const currentMonthEvents = monthEvents.months[fromMonthToIndex(month)];
  return (
    currentMonthEvents &&
    tagSeq.map(currentTag => <td key={currentTag}>{countEventsPerTag(currentTag, currentMonthEvents.events)}</td>)
  );
};

const handleColHover = () => {
  const td = document.querySelectorAll('.summary-details td');

  td.forEach(currentTd => {
    currentTd.addEventListener('mouseover', function() {
      const index = this.cellIndex + 1;
      document.querySelectorAll(`td:nth-child(${index})`).forEach(hoveredTd => {
        hoveredTd.classList.add('highlighted');
      });
    });

    currentTd.addEventListener('mouseleave', function() {
      const index = this.cellIndex + 1;
      document.querySelectorAll(`td:nth-child(${index})`).forEach(hoveredTd => {
        hoveredTd.classList.remove('highlighted');
      });
    });
  });
};

class EventSummary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isSummaryVisible: true,
    };
  }

  toggleSummaryDetails() {
    const { isSummaryVisible } = this.state;
    this.setState({
      isSummaryVisible: !isSummaryVisible,
    });
  }

  render() {
    const { selectedPeriodLevel, grouppedEvents } = this.props;
    const isPeriodLevelSelected = Object.keys(selectedPeriodLevel).length !== 0;
    const activeSummary = this.state.isSummaryVisible ? 'active' : '';
    const buttonText = this.state.isSummaryVisible ? 'HIDE' : 'SHOW';
    return (
      <div className="event-summary">
        <div className="event-header">
          <div className="left-wrapper">
            <span>Overview Events Table</span>
          </div>
          <div className="right-wrapper">
            <button type="button" onClick={() => this.toggleSummaryDetails()}>
              <span>{buttonText}</span>
              <FontAwesomeIcon icon={faChevronRight} />
            </button>
          </div>
          <div className="clear" />
        </div>
        <div className={`summary-details ${activeSummary}`}>
          {handleColHover()}
          <table>
            <tbody>
              <tr className="row-light">
                <th>event tag</th>
                {renderTagIcon()}
              </tr>
              <tr>
                <th>Year</th>
                {renderTagEvents(isPeriodLevelSelected, grouppedEvents[selectedPeriodLevel.year])}
              </tr>
              <tr className="row-light">
                <th>Month</th>
                {renderTagEventsPerMonth(
                  isPeriodLevelSelected && isPeriodLevelSelected.month !== '',
                  selectedPeriodLevel.month,
                  grouppedEvents[selectedPeriodLevel.year],
                )}
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

export default EventSummary;
