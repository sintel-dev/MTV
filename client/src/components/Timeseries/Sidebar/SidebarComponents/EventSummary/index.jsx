import React, { Component } from 'react';
import { tagSeq, fromTagToClassName } from '../../../../Landing/utils';
import { fromMonthToIndex } from '../../../../../model/utils/Utils';

import './EventSummary.scss';

const renderTagIcon = () =>
  tagSeq.map((currentTag) => (
    <td key={currentTag}>
      <span className="tooltip-data">
        <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
        {currentTag}
      </span>
      <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
    </td>
  ));

const countEventsPerTag = (tag, events) => {
  const currentEvents = Object.values(events);
  return currentEvents.filter((currentEvent) => currentEvent.tag === tag).length;
};

const renderTagEvents = (events) =>
  tagSeq.map((currentTag) => <td key={currentTag}>{(events && countEventsPerTag(currentTag, events)) || '-'}</td>);

class EventSummary extends Component {
  componentDidMount() {
    this.handleColHover();
  }

  handleColHover() {
    const td = document.querySelectorAll('.summary-details td');
    td.forEach((currentTd) => {
      currentTd.addEventListener('mouseover', function () {
        const index = this.cellIndex + 1;
        document.querySelectorAll(`td:nth-child(${index})`).forEach((hoveredTd) => {
          hoveredTd.classList.add('highlighted');
        });
      });

      currentTd.addEventListener('mouseleave', function () {
        const index = this.cellIndex + 1;
        document.querySelectorAll(`td:nth-child(${index})`).forEach((hoveredTd) => {
          hoveredTd.classList.remove('highlighted');
        });
      });
    });
  }

  getTimeRangeEvents() {
    const { grouppedEvents, filteredPeriodRange } = this.props;
    const { level } = filteredPeriodRange[0];

    let eventsPerRange = {
      perYear: null,
      perMonth: null,
    };

    if (level === 'month') {
      const year = filteredPeriodRange[0].parent.name;
      if (grouppedEvents[year]) {
        eventsPerRange = {
          perYear: grouppedEvents[year].events,
        };
      }
    }

    if (level === 'day') {
      const year = filteredPeriodRange[0].parent.parent.name;
      const month = filteredPeriodRange[0].parent.name;

      if (grouppedEvents[year]) {
        eventsPerRange = {
          perYear: grouppedEvents[year].events,
          perMonth:
            (grouppedEvents[year].months &&
              grouppedEvents[year].months[fromMonthToIndex(month)] &&
              grouppedEvents[year].months[fromMonthToIndex(month)].events) ||
            null,
        };
      }
    }

    return eventsPerRange;
  }

  showPeriod() {
    const { filteredPeriodRange, selectedPeriodLevel, isTimeSyncEnabled } = this.props;
    let periodString = 'YY/MM';

    if (isTimeSyncEnabled) {
      const { level } = filteredPeriodRange[0];
      if (level === 'month') {
        periodString = periodString.replace('YY', filteredPeriodRange[0].parent.name);
      }

      if (level === 'day') {
        periodString = periodString.replace('YY', filteredPeriodRange[0].parent.parent.name);
        periodString = periodString.replace('MM', filteredPeriodRange[0].parent.name);
      }
    } else {
      if (selectedPeriodLevel.year) {
        periodString = periodString.replace('YY', selectedPeriodLevel.year);
      }
      if (selectedPeriodLevel.month) {
        periodString = periodString.replace('MM', selectedPeriodLevel.month);
      }
    }

    return (
      <div className="period-info">
        <p>{periodString}</p>
      </div>
    );
  }

  render() {
    const { isOpen } = this.props;
    const eventsPerRange = this.getTimeRangeEvents();
    const activeSummary = isOpen ? 'active' : '';

    return (
      <div className="event-summary">
        <div className="event-header">
          <div className="left-wrapper wrapper">
            <span>{this.props.signalName}</span>
          </div>
          <div className="right-wrapper wrapper">{this.showPeriod()}</div>
        </div>
        <div className={`summary-details ${activeSummary}`}>
          <table>
            <tbody>
              <tr>
                <th>event tag</th>
                {renderTagIcon()}
              </tr>
              <tr>
                <th>Year</th>
                {renderTagEvents(eventsPerRange.perYear)}
              </tr>
              <tr>
                <th>Month</th>
                {renderTagEvents(eventsPerRange.perMonth)}
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

export default EventSummary;
