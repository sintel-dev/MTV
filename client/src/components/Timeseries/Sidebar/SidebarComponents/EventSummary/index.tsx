import React, { Component } from 'react';
import * as _ from 'lodash';
import { RootState } from 'src/model/types';
import {
  getDatarunDetails,
  getFilteredPeriodRange,
  getGrouppedDatarunEvents,
  getIsTimeSyncModeEnabled,
  getSelectedPeriodLevel,
} from 'src/model/selectors/datarun';
import { fromMonthToIndex } from 'src/model/utils/Utils';
import { connect } from 'react-redux';
import { tagSeq, fromTagToClassName } from '../../../../Landing/utils';

import './EventSummary.scss';

const renderTagIcon = () =>
  tagSeq.map((currentTag) => (
    <td key={currentTag}>
      <span className="tooltip-data">
        <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
        {currentTag}
      </span>
      <i key={fromTagToClassName(currentTag)} className={`indicator ${fromTagToClassName(currentTag)}`} />
    </td>
  ));

const countEventsPerTag = (filterTag: string, events: object): number => {
  const currentEvents = Object.values(events);
  return currentEvents.filter((currentEvent) => currentEvent.tag === filterTag).length;
};

const renderTagEventsAll = (grouppedEvents) => {
  if (grouppedEvents === undefined) {
    return tagSeq.map((currentTag) => <td key={currentTag}>-</td>);
  }
  return tagSeq.map((currentTag) => {
    let eventSum = 0;
    let eventSet = new Set();
    _.each(grouppedEvents, (value) => {
      _.each(value.events, (event, eventId) => {
        if (event.tag === currentTag) {
          eventSet.add(eventId);
        }
      });
    });
    eventSum = eventSet.size;
    return <td key={currentTag}>{eventSum}</td>;
  });
};

const renderTagEventsPerYear = (periodRange, grouppedEvents) => {
  if (periodRange.level === 'year' || grouppedEvents === undefined) {
    return tagSeq.map((currentTag) => <td key={currentTag}>-</td>);
  }
  return tagSeq.map((currentTag) => <td key={currentTag}>{countEventsPerTag(currentTag, grouppedEvents.events)}</td>);
};

const renderTagEventsPerMonth = (periodRange, month, monthEvents) => {
  if (periodRange.level !== 'day' || monthEvents === undefined) {
    return tagSeq.map((currentTag) => <td key={currentTag}>-</td>);
  }
  const currentMonthEvents = monthEvents.months[fromMonthToIndex(month)];
  if (currentMonthEvents) {
    return tagSeq.map((currentTag) => (
      <td key={currentTag}>{countEventsPerTag(currentTag, currentMonthEvents.events)}</td>
    ));
  }
  return tagSeq.map((currentTag) => <td key={currentTag}>0</td>);
};

type StateProps = ReturnType<typeof mapState>;

class EventSummary extends Component<StateProps> {
  componentDidMount() {
    this.handleColHover();
  }

  handleColHover() {
    const td: NodeList = document.querySelectorAll('.summary-details td');

    td.forEach((currentTd) => {
      currentTd.addEventListener('mouseover', function () {
        const index: number = this.cellIndex + 1;
        document.querySelectorAll(`td:nth-child(${index})`).forEach((hoveredTd) => {
          hoveredTd.classList.add('highlighted');
        });
      });

      currentTd.addEventListener('mouseleave', function () {
        const index: number = this.cellIndex + 1;
        document.querySelectorAll(`td:nth-child(${index})`).forEach((hoveredTd) => {
          hoveredTd.classList.remove('highlighted');
        });
      });
    });
  }

  getTimeRangeEvents() {
    const { grouppedEvents, filteredPeriodRange } = this.props;
    const { level } = filteredPeriodRange[0];

    let eventsPerRange: object = {
      perYear: null,
      perMonth: null,
    };

    if (level === 'month') {
      const year = filteredPeriodRange[0].parent.name;
      if (grouppedEvents[year]) {
        eventsPerRange = {
          perYear: grouppedEvents[year].events,
        };
      }
    }

    if (level === 'day') {
      const year = filteredPeriodRange[0].parent.parent.name;
      const month = filteredPeriodRange[0].parent.name;

      if (grouppedEvents[year]) {
        eventsPerRange = {
          perYear: grouppedEvents[year].events,
          perMonth:
            (grouppedEvents[year].months &&
              grouppedEvents[year].months[fromMonthToIndex(month)] &&
              grouppedEvents[year].months[fromMonthToIndex(month)].events) ||
            null,
        };
      }
    }

    return eventsPerRange;
  }

  render() {
    const { filteredPeriodRange, grouppedEvents } = this.props;

    let currentYear = '';
    let currentMonth = '';
    let currentYearStr = '';
    let currentMonthStr = '';

    let periodRange = filteredPeriodRange[0];
    const { level } = periodRange;
    if (level === 'month') {
      currentYear = periodRange.parent.name;
      currentYearStr = `- ${periodRange.parent.name}`;
    } else if (level === 'day') {
      currentYear = periodRange.parent.parent.name;
      currentYearStr = `- ${periodRange.parent.parent.name}`;
      currentMonth = periodRange.parent.name;
      currentMonthStr = `- ${periodRange.parent.name}`;
    }

    return (
      <div className="event-summary">
        <div className="summary-details">
          <table>
            <tbody>
              <tr>
                <th>event tag</th>
                {renderTagIcon()}
              </tr>
              <tr>
                <th>All</th>
                {renderTagEventsAll(grouppedEvents)}
              </tr>
              <tr>
                <th>Year {currentYearStr} </th>
                {renderTagEventsPerYear(filteredPeriodRange[0], grouppedEvents[currentYear])}
              </tr>
              <tr>
                <th>Month {currentMonthStr}</th>
                {renderTagEventsPerMonth(filteredPeriodRange[0], currentMonth, grouppedEvents[currentYear])}
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

const mapState = (state: RootState) => ({
  selectedPeriodLevel: getSelectedPeriodLevel(state),
  grouppedEvents: getGrouppedDatarunEvents(state),
  filteredPeriodRange: getFilteredPeriodRange(state),
  dataRun: getDatarunDetails(state),
  isTimeSyncEnabled: getIsTimeSyncModeEnabled(state),
});

export default connect<StateProps, {}, {}, RootState>(mapState)(EventSummary);
