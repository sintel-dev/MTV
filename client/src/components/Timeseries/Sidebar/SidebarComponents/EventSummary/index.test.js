import React from 'react';
import EventSummary from './index';
import { grouppedEvents } from '../../../../tests/testmocks/grouppedEvents';

describe('Testing Event Summary component -> ', () => {
  const evtSummaryProps = {
    isSummaryVisible: true,
    selectedPeriodLevel: [],
    grouppedEvents,
  };

  it('Should render without crashing', () => {
    const evtSummaryComponent = shallow(<EventSummary {...evtSummaryProps} />);
    expect(evtSummaryComponent).toMatchSnapshot();
  });

  it('Should display propper button text', () => {
    const evtSummaryComponent = mount(<EventSummary {...evtSummaryProps} />);
    const btnTrigger = evtSummaryComponent.find('#toggleSummary');
    expect(btnTrigger.text()).toContain('HIDE');
    btnTrigger.simulate('click');

    expect(btnTrigger.text()).toContain('SHOW');
  });

  it('Should handle mouse over on column', () => {
    const spy = jest.spyOn(EventSummary.prototype, 'handleColHover');
    const evtSummaryComponent = mount(<EventSummary {...evtSummaryProps} />);
    const tableRow = evtSummaryComponent.find('.summary-details td').first();
    tableRow.simulate('mouseover');

    expect(spy).toBeCalledTimes(1);
  });
});
