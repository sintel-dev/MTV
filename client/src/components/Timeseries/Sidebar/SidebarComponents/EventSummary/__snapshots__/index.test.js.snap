// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Event Summary component ->  Should render without crashing 1`] = `
<div
  className="event-summary"
>
  <div
    className="event-header"
  >
    <div
      className="left-wrapper"
    >
      <span>
        Overview Events Table
      </span>
    </div>
    <div
      className="right-wrapper"
    >
      <button
        id="toggleSummary"
        onClick={[Function]}
        type="button"
      >
        <span>
          HIDE
        </span>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f054",
                "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
              ],
              "iconName": "chevron-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div
      className="clear"
    />
  </div>
  <div
    className="summary-details active"
  >
    <table>
      <tbody>
        <tr
          className="row-light"
        >
          <th>
            event tag
          </th>
          <td
            key="Investigate"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator investigate"
                key="investigate"
              />
              Investigate
            </span>
            <i
              className="indicator investigate"
              key="investigate"
            />
          </td>
          <td
            key="Do not Investigate"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator not_investigate"
                key="not_investigate"
              />
              Do not Investigate
            </span>
            <i
              className="indicator not_investigate"
              key="not_investigate"
            />
          </td>
          <td
            key="Postpone"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator postpone"
                key="postpone"
              />
              Postpone
            </span>
            <i
              className="indicator postpone"
              key="postpone"
            />
          </td>
          <td
            key="Problem"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator problem"
                key="problem"
              />
              Problem
            </span>
            <i
              className="indicator problem"
              key="problem"
            />
          </td>
          <td
            key="Previously seen"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator seen"
                key="seen"
              />
              Previously seen
            </span>
            <i
              className="indicator seen"
              key="seen"
            />
          </td>
          <td
            key="Normal"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator normal"
                key="normal"
              />
              Normal
            </span>
            <i
              className="indicator normal"
              key="normal"
            />
          </td>
          <td
            key="Untagged"
          >
            <span
              className="tooltip-data"
            >
              <i
                className="indicator untagged"
                key="untagged"
              />
              Untagged
            </span>
            <i
              className="indicator untagged"
              key="untagged"
            />
          </td>
        </tr>
        <tr>
          <th>
            Year
          </th>
          <td
            key="Investigate"
          >
            -
          </td>
          <td
            key="Do not Investigate"
          >
            -
          </td>
          <td
            key="Postpone"
          >
            -
          </td>
          <td
            key="Problem"
          >
            -
          </td>
          <td
            key="Previously seen"
          >
            -
          </td>
          <td
            key="Normal"
          >
            -
          </td>
          <td
            key="Untagged"
          >
            -
          </td>
        </tr>
        <tr
          className="row-light"
        >
          <th>
            Month
          </th>
          <td
            key="Investigate"
          >
            -
          </td>
          <td
            key="Do not Investigate"
          >
            -
          </td>
          <td
            key="Postpone"
          >
            -
          </td>
          <td
            key="Problem"
          >
            -
          </td>
          <td
            key="Previously seen"
          >
            -
          </td>
          <td
            key="Normal"
          >
            -
          </td>
          <td
            key="Untagged"
          >
            -
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
